jbmc/regression/jbmc-strings/StringBufferSetLength/SetLengthTest.java

```diff
@@ -0,0 +1,30 @@
+public class SetLengthTest {
+
+    public static void main(int unknown) {
+
+        StringBuffer s = new StringBuffer("ABCD");
+        // Avoid producing a huge string that consumes too much memory, or which
+        // string refinement can't handle:
```

Comment:
:information_source: string refinement could handle a huge string there, but concretizing it in a trace would consume too much memory so it would get replaced by nil (which would make an invariant fail in trace checking)